int32 side

int32 LEFT_TAP=1
int32 RIGHT_TAP=2
---
---
